test/unit/Router/RouteLoaderTest.php

```diff
@@ -0,0 +1,213 @@
+            ],
+            [
+                'resource' => self::SPECIFICATION_DEFAULT_NAME,
+                'type' => RouteLoader::OPENAPI_SPEC,
+            ],
+            [
+                'resource' => self::SPECIFICATION_DEFAULT_NAME,
+                'type' => RouteLoader::OPENAPI_OPERATION,
+            ],
+        ];
+    }
+
+    /**
+     * @param mixed $resource
+     *
+     * @throws Throwable
+     *
+     * @dataProvider loadProvider
+     */
+    public function testLoad($resource, ?string $type = null): void
+    {
+        $operations = [
+            'first' => $this->operationMock,
+            'second' => $this->operationMock,
+        ];
+
+        $requestParameters = [
+            'path' => new ObjectType([new Property('path')]),
+        ];
+
+        $operationUrl    = 'https://custom.local';
+        $operationMethod = 'customAction';
+        $requirements    = [];
+
+        $expectedOperationsCount = count($operations);
+
+        $this->specificationLoaderMock
+            ->expects(self::once())
+            ->method('load')
+            ->willReturn($this->specificationMock);
+
+        $this->specificationMock
+            ->expects(self::once())
+            ->method('getOperations')
+            ->willReturn($operations);
+
+        $this->operationMock
+            ->expects(self::exactly($expectedOperationsCount))
+            ->method('getRequestParameters')
+            ->willReturn($requestParameters);
+
+        $this->argumentResolverMock
+            ->expects(self::exactly($expectedOperationsCount))
+            ->method('resolveArgumentPatterns')
+            ->with($requestParameters['path'])
+            ->willReturn($requirements);
+
+        $this->operationMock
+            ->expects(self::exactly($expectedOperationsCount))
+            ->method('getUrl')
+            ->willReturn($operationUrl);
+
+        $this->operationMock
+            ->expects(self::exactly($expectedOperationsCount))
+            ->method('getMethod')
+            ->willReturn($operationMethod);
+
+        $routeLoader = new RouteLoader(
+            $this->specificationLoaderMock,
+            $this->argumentResolverMock
+        );
+
+        $routeCollection = $routeLoader->load($resource, $type);
+
+        foreach ($operations as $operationId => $operation) {
+            $route = $routeCollection->get($operationId);
+
+            if ($route === null) {
+                continue;
+            }
+
+            Assert::assertSame(
+                'OnMoon\OpenApiServerBundle\Controller\ApiController::handle',
+                $route->getDefault('_controller')
+            );
+            Assert::assertSame(
+                (string) $resource,
+                $route->getOption(RouteLoader::OPENAPI_SPEC)
+            );
+            Assert::assertSame(
+                $operationId,
+                $route->getOption(RouteLoader::OPENAPI_OPERATION)
+            );
```

Comment:
We should also verify the correctness of:
- $route->getPath()
- $route->getRequirements() (with a non-empty and empty list of requirements)
- $route->getSchemes(), $route->getHost() (verify they are empty)
- $route->getMethods()